executor/cook/executor.py

```diff
@@ -99,34 +99,34 @@ def update_status(self, task_state, reason=None):
                 return False
 
 
-def send_message(driver, message, max_message_length):
+def send_message(driver, message):
     """Sends the message, if it is smaller than the max length, using the driver.
 
+    Note: This function must rethrow any OSError exceptions that it encounters.
+
     Parameters
     ----------
     driver: MesosExecutorDriver
         The driver to send the message to.
-    message: object
+    message: dictionary
         The raw message to send.
-    max_message_length: int
-        The allowed max message length after encoding.
 
     Returns
     -------
     whether the message was successfully sent
     """
-    logging.info('Sending framework message {}'.format(message))
-    message_string = str(message).encode('utf8')
-    if len(message_string) <= max_message_length:
+    try:
+        logging.info('Sending framework message {}'.format(message))
+        message_string = json.dumps(message).encode('utf8')
         encoded_message = pm.encode_data(message_string)
         driver.sendFrameworkMessage(encoded_message)
         return True
-    else:
-        log_message_template = 'Unable to send message as its length of {} exceeds allowed max length of {}'
-        logging.warning(log_message_template.format(len(message_string), max_message_length))
+    except OSError as os_error:
+        raise os_error
+    except Exception:
+        logging.exception('Error in sending message {}'.format(message))
```

Comment:
Were these new `except` clauses for a reason?

Reply:
I added the exception handling to allow the executor to chug along even if there is an error in sending a framework message. The exception handler needs to rethrow the `OSError` so that cleanup can be performed higher up.

Reply:
FYI, you can leave the exception value anonymous if all you want to with it do is re-raise it:
```python
except OSError:
    raise
```

Reply:
Done.